pageserver/src/walredo.rs

```diff
@@ -584,27 +582,168 @@ struct PostgresRedoProcess {
     stdin: ChildStdin,
     stdout: ChildStdout,
     stderr: ChildStderr,
-    called_kill: bool,
+    wal_redo_timeout: Duration,
+    // Ring buffer with channel identifiers of buffered redo requests
+    ring_buf: [Option<Sender<Bytes>>; N_CHANNELS],
+    // Head position in ring buffer
+    ring_buf_pos: usize,
+    // Number of buffered requests
+    n_buffered: usize,
```

Comment:
Hmm, so the ring buffer is now used just locally in the thread, to keep track which response received from the walredo process belongs to which request. Let's replace it with `std::collections::VecDeque`

Reply:
Done